### DISCUSSION TITLE
Rarity Data output

### DISCUSSION THREAD

**Discussion by zWavey** [nftchef/art-engine#27] [Help + Q&A]
Is there any way I am able to out put data on the rarities after the collection is made. For example like hashlips beta feature screenshot below
![Screenshot 2022-02-10 133039](https://user-images.githubusercontent.com/99408191/153418069-36b94762-65b5-4abf-9a59-050d587837fa.png)

**Comment by mnedelko** [comment]
I would like to bump this comment. This would be tremendously useful 🙇

**Comment by nftchef** [comment]
hi @zWavey , @mnedelko 

I'm in the process of rewriting the rarity algorithm and output, the latest rarity.js script is functional and exists on the https://github.com/nftchef/art-engine/tree/utils/rarity branch.

Work in progress documentation for it is also, here, https://0n10ndevision.gitbook.io/nftchef-art-engine/utils/rarity

**Reply by developmentd3** [reply]
@nftchef could you drop a quick hint about any reason why the _ranking wouldn't be populating after running the latest rarity.js?

**Reply by nftchef** [reply]
do you have an error? and when you say latest, are you on the WIP rarity branch? or the main `/nested-folder-structure` branch? the latest main branch is still broken

**Reply by developmentd3** [reply]
I was just copying the WIP update from rarity branch utils/rarity.js and using it in the main. Was that the mistake?

**Reply by nftchef** [reply]
That _should_ work if you copied the new file and called it.
it does require `csv-writer` which would not have been installed if you copied it manually.
```
yarn add csv-writer

// if using npm
npm install --save csv-writer
```

**Reply by developmentd3** [reply]
Thanks for the Quick update, I managed to fix that csv-writer gap and funny thing is it does generate the _rarity.csv, but not the expected _ranking.csv. i'll continue to debug and see if i catch anything

**Reply by BadGenius22** [reply]
> hi @zWavey , @mnedelko
> 
> I'm in the process of rewriting the rarity algorithm and output, the latest rarity.js script is functional and exists on the https://github.com/nftchef/art-engine/tree/utils/rarity branch.
> 
> Work in progress documentation for it is also, here, https://0n10ndevision.gitbook.io/nftchef-art-engine/utils/rarity

Hi couldn't find rarity file there.
Are you still in the process of rewriting?

**Reply by AlHashemiUAE** [reply]
> Thanks for the Quick update, I managed to fix that csv-writer gap and funny thing is it does generate the _rarity.csv, but not the expected _ranking.csv. i'll continue to debug and see if i catch anything

Were you able to make it work and create `_ranking.csv` together with `_rarity.csv` ?

**Reply by AlHashemiUAE** [reply]
> > Thanks for the Quick update, I managed to fix that csv-writer gap and funny thing is it does generate the _rarity.csv, but not the expected _ranking.csv. i'll continue to debug and see if i catch anything
> 
> Were you able to make it work and create `_ranking.csv` together with `_rarity.csv` ?

Found a FIX!!!!! Simply change this code:

```
// console.log(tokenRarities);
  console.table(tokenRarities);
  options.outputRanking ? outputRankingCSV(tokenRarities) : null;
}
```

to this: 

```
  console.log(tokenRarities);
  console.table(tokenRarities);
  options.outputRanking ? true : outputRankingCSV(tokenRarities);
```
Enjoy :)

**Comment by BadGenius22** [comment]
Hi @nftchef , I have an error while try to run "npm run rarity" command.
Got error message: **Missing script: "rarity"**

I already installed 
`// if using npm
npm install --save csv-writer`


<img width="598" alt="Screen Shot 2022-04-26 at 18 13 03" src="https://user-images.githubusercontent.com/103027181/165287926-b03cbd11-4cd5-4fc3-ab23-d6b885676abe.png">
